demonstrations/qsim_supremacy.py

```diff
@@ -0,0 +1,567 @@
+#
+
+m = 14  # number of cycles
+
+# gate_sequence = np.resize(["A", "B", "C", "D", "C", "D", "A", "B"], m)
+gate_sequence = np.resize(["A", "B", "C", "D"], m)
+
+
+######################################################################
+# The single-qubit gates are randomly selected and applied to each qubit in
+# the circuit, while avoiding the same gate being applied to the same wire
+# twice in a row. We do this by creating a helper function that
+# specifies the order in which the single-qubit
+# gates should be applied. We can use this list within the
+# circuit to know which gate to apply when.
+#
+
+def generate_single_qubit_gate_list():
+    # create the first list by randomly selecting indices
+    # from `single_qubit_gates`
+    g = [list(np.random.choice(range(len(single_qubit_gates)), size=wires))]
+
+    for cycle in range(len(gate_sequence)):
+        g.append([])
+        for w in range(wires):
+            # check which gate was applied to the wire previously
+            one_gate_removed = list(range(len(single_qubit_gates)))
+            bool_list = np.array(one_gate_removed) == g[cycle][w]
+
+            # and remove it from the choices of gates to be applied
+            pop_idx = np.where(bool_list)[0][0]
+            one_gate_removed.pop(pop_idx)
+            g[cycle + 1].append(np.random.choice(one_gate_removed))
+    return g
+
+
+######################################################################
+# Finally, we can define the circuit itself and create a QNode that we will
+# use for circuit evaluation with the ``qsim`` device. The two-qubit gates
+# are applied to the qubits connected by A, B, C, or D as defined above.
+# The circuit ends with a half-cycle, consisting of only a layer of
+# single-qubit gates.
+#
+# From the QNode, we also need both the probabilities of the measurement
+# results, as well raw samples. To facilitate this, we add a keyword
+# argument to our circuit allowing us to switch between the two returns. We
+# sample from the Pauli-Z observable on all wires, which will give us the
+# eigenvalues :math:`\pm 1` of the observable, corresponding to the states
+# :math:`\left|0\right>` and :math:`\left|1\right>`.
+#
+
+@qml.qnode(dev)
+def circuit(seed=42, return_probs=False):
+    np.random.seed(seed)
+    gate_idx = generate_single_qubit_gate_list()
+
+    # m full cycles - single-qubit gates & two-qubit gate
+    for i, gs in enumerate(gate_sequence):
+        for w in range(wires):
+            single_qubit_gates[gate_idx[i][w]](wires=w)
+
+        for qb_1, qb_2 in gate_order[gs]:
+            ops.ISWAP(wires=(qb_1, qb_2))
+            ops.CPhase(-np.pi/6, wires=(qb_1, qb_2))
+
+    # one half-cycle - single-qubit gates only
+    for w in range(wires):
+        single_qubit_gates[gate_idx[-1][w]](wires=w)
+
+    if return_probs:
+        return qml.probs(wires=range(wires))
+    else:
+        return [qml.sample(qml.PauliZ(i)) for i in range(wires)]
```

Comment:
I like this!

Reply:
Do we want to make it PEP8 compliant, e.g.,
```suggestion
    if return_probs:
        return qml.probs(wires=range(wires))

    return [qml.sample(qml.PauliZ(i)) for i in range(wires)]
```

I'm in two minds, because an explicit if-else makes it easier for readers to follow.

Reply:
I vote no. I actually find explicit `if/else` more readable

Reply:
I think I agree

Reply:
Thanks. I've started following this PEP8 suggestion recently (and I've actually started to like it), although I don't mind either way to be honest. I'll keep it like this then since there are 2 no and 1 abstain. :laughing: